### DISCUSSION TITLE
Get BaseAddress in api extension

### DISCUSSION THREAD

**Discussion by netclectic** [canton7/RestEase#186] [Q&A] [answered]
Is it possible to get hold of the BaseAddress of the RestClient from within an api extension method?

I have an api extension method that needs to pass a header parameter that is an absolute url, but my calling method has no clue about that configuration detail. My extension method has enough detail to construct the url if it can get hold of the base address, but I can't see any obvious way to get it :-/

Is there some magic property like the IRequester that I could add to the api interface?

**Comment by canton7** [comment, marked as answer]
Unfortunately not: the base address could be coming from the HttpClient, or from attributes on the interface, or a combination of those.

It might be easier to write a RequestModifier? You could use a custom attribute to tell the RequestModifier it needs to add this header.

**Reply by canton7** [reply]
I don't think I'd be adverse to adding a HttpClient property to IRequester either.

**Reply by netclectic** [reply]
I can see how having access to the HttpClient from the IRequester would be useful, but it feels kind of dirty exposing the IRequester on the api interface.

I'll try and get the RequestModifier approach working, I should be able to set the header with a relative url in the extension method and get the RequestModified to apply that base address to it.

Thanks!

**Reply by canton7** [reply]
These things might help you:

 - [Adding to HttpRequestMessage.Properties](https://github.com/canton7/RestEase#adding-to-httprequestmessageproperties) -- if you need a way to pass data down to a RequestModifier
 - You can [fetch the `RequestInfo` from HttpRequestMessage.Properties](https://github.com/canton7/RestEase#requestmodifier), and from there you can get the `MethodInfo` of the method which was invoked, which you can use to fetch any custom attributes you added.

**Reply by netclectic** [reply]
I was already using a custom http client handler to implement Polly retry policies and set a proxy, I was able to do what I needed in there.

**Reply by canton7** [reply]
Glad to hear! You can chain HttpMessageHandlers (with DelegatingHandler) if that makes things neater.
